src/core/inc/Defines.h

```diff
@@ -50,6 +50,14 @@
 //! This define is deprecated.  Remove any #ifdef statements in user code.
 #define QUESO_EXPECTS_LN_LIKELIHOOD_INSTEAD_OF_MINUS_2_LN
 
+// Use GSL inline functions
+#define HAVE_INLINE
+
+// And only do GSL range-checking if we're really debugging
+#ifndef DEBUG
```

Comment:
Shouldn't this be `NDEBUG`?  Or at least `#ifdef DEBUG`.  Please look up which of these patterns we're using, because I am not entirely certain.